src/main/java/org/teraslogy/simpleliquids/block/entity/LiquidEntitySystem.java

```diff
@@ -143,4 +148,44 @@ public void update(float delta) {
             timeSinceLastUpdate -= 0.3f;
         }
     }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL)
+    public void onChunkGenerated(OnChunkGenerated chunkGenerated, EntityRef entity) {
+        chunkWorker(chunkGenerated.getChunkPos(), entity);
+    }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL)
+    public void onChunkLoaded(OnChunkLoaded chunkLoaded, EntityRef entity) {
+        chunkWorker(chunkLoaded.getChunkPos(), entity);
+    }
+
+    public void chunkWorker(Vector3i chunkPos, EntityRef entity) {
```

Comment:
I would name it something else, mabye `processChunk` or similar. With `chunkWorker` I expect some worker class to be returned, which is not the case here